cloud/wfm-rest-api/src/data-api/CrudRepository.ts

```diff
@@ -0,0 +1,51 @@
+import * as Promise from 'bluebird';
+import { Db } from 'mongodb';
+import { SortedPageRequest } from '../data-api/PageRequest';
+import { PageResponse } from '../data-api/PageResponse';
+
+/**
+ * Interface for building CRUD database repository.
+ * Clients should implement this interface to connect storage of WFM objects
+ * with different data storage engines like MySQL etc.
+ * Interface is being used internally to perform database operations for WFM models
+ * It's not recomended to use it for other application business logic.
+ */
+export interface CrudRepository {
```

Comment:
I don't like using the term CRUD itself, feels like too much slang and signals boilerplate code.

Could be `PagingDataRepository` or something, much fancier :D